03_Modules/OcppRouter/src/module/router.ts

```diff
@@ -34,20 +23,25 @@ export class MessageRouterImpl extends AbstractMessageRouter implements IMessage
     protected _handler: IMessageHandler;
     protected _networkHook: (identifier: string, message: string) => Promise<boolean>;
 
-    private _onConnectionCallbacks: ((identifier: string, info?: Map<string, string>) => Promise<boolean>)[] = [];
-    private _onCloseCallbacks: ((identifier: string, info?: Map<string, string>) => Promise<boolean>)[] = [];
-    private _onMessageCallbacks: ((identifier: string, message: string, info?: Map<string, string>) => Promise<boolean>)[] = [];
-    private _sentMessageCallbacks: ((identifier: string, message: string, error?: any, info?: Map<string, string>) => Promise<boolean>)[] = [];
+    // Structure of the maps: key = identifier, value = array of callbacks
```

Comment:
Just to be sure here, and this is probably worth a comment:
Since web socket connection are stateful and tied to a single instance, we can put the callbacks into memory. And when a charger connects, it gets the callbacks from the DB and loads them into memory.
 
A small thing which probably isn't worth a deeper dive right now, but when sending the request to set up a callback, we need to be sure to know where the charger is connected to. Else, we could hit instance `A` but the charger is connected to instance `B` and since the callback state only gets refreshed on connection, instance `B` won't know about this until the charger drops and reconnects. But I think this is down the line.

Reply:
Good point. I added a comment to the AdminApi class. I'm not sure what the correct way around this is, but I also agree we can leave it until later to fix.